File: apps/ios/SofaExchange/Domain/UseCases/SearchListingsUseCase.swift
final class SearchListingsUseCase {
    private let repository: ListingsRepositoryProtocol

    init(repository: ListingsRepositoryProtocol) {
        self.repository = repository
    }

    func execute(query: SearchQuery) async throws -> [Listing] {
        try await repository.search(query: query)
    }
}

File: apps/android/app/src/main/java/com/sofaexchange/domain/usecase/SearchListingsUseCase.kt
package com.sofaexchange.domain.usecase

import com.sofaexchange.domain.model.Listing
import com.sofaexchange.domain.repository.ListingsRepository
import com.sofaexchange.domain.repository.SearchParams

class SearchListingsUseCase(private val repository: ListingsRepository) {
    suspend fun execute(params: SearchParams): List<Listing> = repository.search(params)
}
